modules/gdscript/gdscript_parser.cpp

```diff
@@ -5062,6 +5062,31 @@ void GDScriptParser::_determine_inheritance(ClassNode *p_class) {
 					return;
 				}
 				p = NULL;
+			} else {
+				List<PropertyInfo> props;
+				ProjectSettings::get_singleton()->get_property_list(&props);
+				for (List<PropertyInfo>::Element *E = props.front(); E; E = E->next()) {
```

Comment:
I'm a bit concerned about doing this project settings parsing all the time, it sounds like it could be quite heavy.
I see that existing code seems to do it the same way, with quite a bit of code duplication... couldn't we factor this out and cache the results somehow? CC @vnen @bojidar-bg

Reply:
We do need a central authority for getting autoloads. But for now this is the only way since non-tool autoloads aren't available in the editor, and they aren't listed anywhere else.

Reply:
Created an issue about this particular problem: #31899